packages/docusaurus-theme-classic/src/theme/DocSidebar/index.js

```diff
@@ -176,6 +177,13 @@ function DocSidebar(props) {
   } = props;
 
   useLockBodyScroll(showResponsiveSidebar);
+  const windowSize = useWindowSize();
+
+  useEffect(() => {
+    if (windowSize === windowSizes.desktop && showResponsiveSidebar) {
```

Comment:
here too you could remove `showResponsiveSidebar ` (or you need to include it in useEffect deps)